.azure-pipelines/documentation_deploy.yml

```diff
@@ -0,0 +1,75 @@
+parameters:
+  - name: release
+    type: boolean
+    default: False
+
+jobs:
+  - job: "documentation"
+    displayName: "Documentation"
+    pool:
+      vmImage: "ubuntu-latest"
+    variables:
+      documentation_artefact_path: "$(Build.StagingDirectory)/documentation"
+      documentation_repo_path: "$(Build.StagingDirectory)/git"
+    steps:
+      - task: DownloadBuildArtifacts@0
+        inputs:
+          buildType: "current"
+          specificBuildWithTriggering: true
+          downloadType: "single"
+          artifactName: "documentation"
+          downloadPath: "$(Build.StagingDirectory)"
+        displayName: "Retrieve updated documentation"
+      - task: InstallSSHKey@0
+        inputs:
+          knownHostsEntry: "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ=="
+          sshPublicKey: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCh4HFo2vH2N+skVC7NDURwSTdoTCPLFyasbb9ijUzxrBgSIfYBaktX4c/dEq/V9IUWd7d1U1UOjHSNVN2hjk/pOVPjMdTA06xbzX8VhF/paS2MLKTSf72AQQ2jH/CBYmwypYCIpg+9Xdnne4pqENQsDtuZqTuGD2git3UBIAStnSHDLsJedu5Q38nYNQ5XJxpoU0Yu2A1+a2aRKLTd7ZlW9BmWEM+nQoe86tdGjJkHqXDSBLnmePdc/jzFibkK69b/lXGQrpf4IahOAAwJdm5uohAZF03l6ySUETrpSTZRKXrczfZ0NJ+eF6/TkLqW8oh2GktRGaYQ8zuKGACvMVKN"
+          sshKeySecureFile: "azure-ess-notebooks-key"
+        displayName: "Add GitHub Pages deploy key"
+      - bash: |
+          set -ex
+          # Publishing documentation involves a Git commit, this sets the identity for that commit
+          git config --global user.name 'The Great Documentation Generation Entity'
+          git config --global user.email 'nobody@localhost'
+        displayName: "Git config"
+      - bash: |
+          set -ex
+          git clone git@github.com:scipp/ess-notebooks "$(documentation_repo_path)"
+        displayName: "Clone documentation repository"
+      - bash: |
+          set -ex
+          cd "$(documentation_repo_path)"
+          # checkout the gh-pages branch
+          git checkout gh-pages
+          # Stage removal of everything
+          git rm -rf .
+          git status
+        displayName: "Remove old documentation, keeping releases"
+      - bash: |
+          set -ex
+          cd "$(documentation_repo_path)"
+          # The .nojekyll file is used to turn off GitHub Pages building
+          touch .nojekyll
+          rsync -av "$(documentation_artefact_path)/" .
+          git add .
+          git status
+        displayName: "Copy new documentation"
+      - bash: |
+          set -ex
```

Comment:
Are these release docs currently created everytime this template is run?

Reply:
Yes, do you suggest otherwise?

Reply:
I think this means we will create a new set of release docs every-time master is updated. If we are planning to keep these around this could quickly lead to quite a lot of doc builds in the releases folder. 

The main scipp repo handles this by only creating permanent release doc builds which are not overwritten when the release.yml pipeline is run. https://github.com/scipp/scipp/blob/master/.azure-pipelines/release.yml

Reply:
Actually think I have missed something here as above we don't preserve previous release docs anyway and I missed the condition on line 65.